app/src/org/commcare/utils/FileUtil.java

```diff
@@ -632,19 +651,23 @@ public static void updateFileLocationFromIntent(Context context, Intent intent,
 
     // get a filePath from an intent returned from file provider
     @Nullable
-    public static String getFileLocationFromIntent(Intent intent) {
+    public static String getFileLocationFromIntent(Context context, Intent intent) {
         // Android versions 4.4 and up sometimes don't return absolute
         // filepaths from the file chooser. So resolve the URI into a
         // valid file path.
         Uri uriPath = intent.getData();
         if (uriPath != null) {
-            String filePath;
+            String fileName = getFileName(context.getContentResolver(), uriPath);
+            File destinationFile = new File(context.getCacheDir(), fileName);
             try {
-                filePath = UriToFilePath.getPathFromUri(CommCareApplication.instance(), uriPath);
-            } catch (UriToFilePath.NoDataColumnForUriException e) {
-                filePath = uriPath.toString();
+                InputStream inputStream = context.getContentResolver().openInputStream(uriPath);

```

Comment:
we should not need to make an unnecessary file copy here as consumers of result of this function like the `UnzipTask` already makes [that copy for internal processing](https://github.com/dimagi/commcare-android/blob/master/app/src/org/commcare/tasks/UnzipTask.java#L67). We should just return the uri from here and have callers of this method always work on a uri instead of the path (all these paths should already have uri handling since this function is already expected to return uri in case of errors during file path resolution).